### ISSUE TITLE
Skinned Geometry Support: black t-rex | Issues with Dynamic scenes

### ISSUE THREAD

**Issue by JerryI** [gkjohnson/three-gpu-pathtracer#588]
**Describe the bug**

It seems that Dynamic scene generation is not working on Mac M1/M2 & Safari, Firefox

[__Skinned Geometry__](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/skinnedMesh.html)

<img width="968" alt="Screenshot 2024-03-24 at 23 04 04" src="https://github.com/gkjohnson/three-gpu-pathtracer/assets/4111822/94192f8d-4639-4cbb-b6ea-c7e4c0aa8926">

It is almost black (there is still some profile of a dinosaur visible ), while in the console

<img width="981" alt="Screenshot 2024-03-24 at 23 04 37" src="https://github.com/gkjohnson/three-gpu-pathtracer/assets/4111822/42122cd6-e643-4ab0-a82c-f13d96a4f188">

**Expected behavior**

A walking dinosaur 

**Extra**

Also once I tried to change in my project from `PathTracingSceneGenerator` to dynamic, however, it immediately generates an error

<img width="968" alt="Screenshot 2024-03-24 at 23 06 27" src="https://github.com/gkjohnson/three-gpu-pathtracer/assets/4111822/195ba7fb-ed40-42cb-956c-058e447e49f6">

About 1-2 years ago with a legacy version of Three.js and your library it did work great

**Platform:**

 - Device: Desktop
 - OS: MacOS
 - GPU: M1
 - Browser: Firefox
 - Browser Version: 123.0.1
 - Three.js version: ^0.160.0
 - Library version: ^0.0.16

**Comment by gkjohnson** [comment]
I'll probably need some help digging into what these errors mean and what might be causing it if you'd like to take a look. cc @robertoranon since you were looking into some floating point texture mismatches.

The project is also going through a pretty big rework at the moment so it's likely we'll need to retest some of these platforms once it's finished.

**Comment by robertoranon** [comment]
sure, happy to help and make tests!

**Comment by JerryI** [comment]
> I'll probably need some help digging into what these errors mean and what might be causing it if you'd like to take a look. cc @robertoranon since you were looking into some floating point texture mismatches.
> 
> The project is also going through a pretty big rework at the moment so it's likely we'll need to retest some of these platforms once it's finished.

Your project is amazing. It is just some minor complains from me ;))

 I thought that may be the dynamic demos are rearely used and after the refactoring and updates some stuff got broken.

If it helps, i can provide more details and traces from the console

**Comment by gkjohnson** [comment]
> sure, happy to help and make tests!
> ...
> I get the same problem with Firefox also on index.html and other examples ... @JerryI which examples do you see working on Firefox?

Thanks! From the warnings it looks like it's more half float / float texture incompatibility? Maybe we need to check for more extensions? Not immediately clear to me.

**Comment by JerryI** [comment]
Ok, it turned out to have a more general issue. Not only dynamics

1. [Basic gLTF](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/basic.html) Safari ✅ Firefox ✅
2. [Geometry Primitives](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/primitives.html) Safari ✅ Firefox ✅
3. [Phys Based materials](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/index.html)

Safari ✅ 

Firefox ❌

![Screenshot 2024-03-26 at 17 49 47](https://github.com/gkjohnson/three-gpu-pathtracer/assets/4111822/c6063b41-e9c4-4a68-8b01-cb2ff9831a00)

console output

```bash
This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”.
[index.html](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/index.html)
THREE.WebGLProgram: Program Info Log: WARNING: Output of vertex shader 'webgl_37cc066937ccaa46' not read by fragment shader [three.module.js:20290:12](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/node_modules/three/build/three.module.js)
WebGL warning: texSubImage: Alpha-premult and y-flip are deprecated for non-DOM-Element uploads.
WebGL warning: readPixels: Format and type RGBA/HALF_FLOAT incompatible with this RGBA16F attachment. This framebuffer requires either RGBA/FLOAT or getParameter(IMPLEMENTATION_COLOR_READ_FORMAT/_TYPE) RGBA/FLOAT.
```

5. [Lego](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/lego.html) 
 
Safari ✅ 

Firefox ❌

![Screenshot 2024-03-26 at 17 52 59](https://github.com/gkjohnson/three-gpu-pathtracer/assets/4111822/763a6a0b-b6b4-4aed-8d45-860b06968c9f)

console output

```bash
This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”.
[lego.html](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/lego.html)
THREE.WebGLProgram: Program Info Log: WARNING: Output of vertex shader 'webgl_37cc066937ccaa46' not read by fragment shader [three.module.js:20290:12](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/node_modules/three/build/three.module.js)
WebGL warning: generateMipmap: Tex image TEXTURE_2D level 0 is incurring lazy initialization.
WebGL warning: texSubImage: Alpha-premult and y-flip are deprecated for non-DOM-Element uploads.
WebGL warning: readPixels: Format and type RGBA/HALF_FLOAT incompatible with this RGBA16F attachment. This framebuffer requires either RGBA/FLOAT or getParameter(IMPLEMENTATION_COLOR_READ_FORMAT/_TYPE) RGBA/FLOAT.
WebGL warning: generateMipmap: Tex image TEXTURE_2D level 0 is incurring lazy initialization.
```

8. [Interior](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/interior.html)  Safari ✅ Firefox ✅ WOW!


10. [DOF](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/depthOfField.html)  

Safari ✅ 

Firefox ❌

Models are missing

![Screenshot 2024-03-26 at 17 55 05](https://github.com/gkjohnson/three-gpu-pathtracer/assets/4111822/dae0042c-5c3a-48f6-9cfe-91ee7ab7799d)

console

```bash
This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”.
[depthOfField.html](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/depthOfField.html)
THREE.WebGLProgram: Program Info Log: WARNING: Output of vertex shader 'webgl_37cc066937ccaa46' not read by fragment shader [three.module.js:20290:12](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/node_modules/three/build/three.module.js)
WebGL warning: texSubImage: Alpha-premult and y-flip are deprecated for non-DOM-Element uploads.
WebGL warning: readPixels: Format and type RGBA/HALF_FLOAT incompatible with this RGBA16F attachment. This framebuffer requires either RGBA/FLOAT or getParameter(IMPLEMENTATION_COLOR_READ_FORMAT/_TYPE) RGBA/FLOAT.
```






12. [Skinned Geometry](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/skinnedMesh.html)

__Safari__ ✅
Looks like if it works

![Screenshot 2024-03-26 at 17 42 58](https://github.com/gkjohnson/three-gpu-pathtracer/assets/4111822/0cf711cc-f75e-46d4-8f04-37276a9dfd88)

JS console is empty

__Firefox__ ❌
Black screen

![Screenshot 2024-03-26 at 17 45 44](https://github.com/gkjohnson/three-gpu-pathtracer/assets/4111822/7f2b5df9-a60d-4185-b07c-9fc0c095bf80)

console output

```bash
This page is in Quirks Mode. Page layout may be impacted. For Standards Mode use “<!DOCTYPE html>”.
[skinnedMesh.html](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/skinnedMesh.html)
THREE.WebGLProgram: Program Info Log: WARNING: Output of vertex shader 'webgl_37cc066937ccaa46' not read by fragment shader [three.module.js:20290:12](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/node_modules/three/build/three.module.js)
WebGL warning: texSubImage: Alpha-premult and y-flip are deprecated for non-DOM-Element uploads.
WebGL warning: readPixels: Format and type RGBA/HALF_FLOAT incompatible with this RGBA16F attachment. This framebuffer requires either RGBA/FLOAT or getParameter(IMPLEMENTATION_COLOR_READ_FORMAT/_TYPE) RGBA/FLOAT.
WebGL warning: drawArraysInstanced: Using format enabled by implicitly enabled extension: EXT_float_blend. For maximal portability enable it explicitly. 30
After reporting 32, no further warnings will be reported for this WebGL context.
```

13. [Morph](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/skinnedMesh.html#morphtarget)

__Safari__ works ✅

__Firefox__ black ❌

![Screenshot 2024-03-26 at 17 48 02](https://github.com/gkjohnson/three-gpu-pathtracer/assets/4111822/7764973b-1c91-4492-bf97-82102584675d)

14. [Spot Light](https://gkjohnson.github.io/three-gpu-pathtracer/example/bundle/areaLight.html) Safari ✅ Firefox ✅

**Comment by gkjohnson** [comment]
Can you check if these extensions are supported in the problem browser:

```
OES_texture_float_linear
EXT_color_buffer_half_float
EXT_color_buffer_float
EXT_float_blend
```

You can see if they're listed on this site:

https://webglreport.com/?v=2

**Comment by JerryI** [comment]
sure 
[WebGL Report.pdf](https://github.com/gkjohnson/three-gpu-pathtracer/files/14887242/WebGL.Report.pdf)

**Comment by gkjohnson** [comment]
This should be fixed now in #623. It seems that firefox does not support reading half float data back from the GPU.
